### PR TITLE
changed: handle tuples in eclmpiserializer

### PR DESCRIPTION
**Pull request by akva2**
thus we can remove support in MPIPacker

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: ebos/eclmpiserializer.hh [hunk 1]
```diff
@@ -498,6 +524,20 @@ protected:
           variadic_call(std::forward<Args>(args)...);
     }
 
+    template<std::size_t I = 0, typename Tuple>
+    typename std::enable_if<I == std::tuple_size<Tuple>::value, void>::type
```
Comment (blattms) [inline ebos/eclmpiserializer.hh:528]: Now I wonder why we don't treat pairs like tuple. :laughing: They should be accessible just like these.

**Review by blattms — APPROVED** [review]
Looks correct.

File: ebos/eclmpiserializer.hh (hunk 1 above)
Comment (akva2) [reply ebos/eclmpiserializer.hh:528]: hmm, that's a good point. i'll see if i can make that work.

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
@blattms please re-review (i added tuple as pair)..

**Review by blattms — COMMENTED** [review]
As in the other PR i now wonder whether we should have a specialization of `is_tuple` for pair and use that like
```
-  } else if constexpr (is_pair<T>::value) {
+ } else if constexpr (is_tuple<T>::value) {
-            pair(data);
+            tuple(data);
```
Other than that this looks great.

**Comment by akva2** [comment]
okay, i have combined the predicates for pair and tuples.

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
i'll merge this now, comments are addressed.

**Pull request merged**